### PR TITLE
Add ability to add multiple sheets to a workbook

### PR DESCRIPTION
**Pull request by elliotstoner**
There have been a few previous PRs for adding multi-sheet functionality - this is actually built off of [davidryansoftware's fork](https://github.com/davidryansoftware/xlsxfromjson), but with a few tweaks. I'd love feedback and suggestions, and hopefully we can finally get this merged in!

Some key things from feedback previously given:
- File size increases 28% from 6.0KB to 7.7KB for module.js; 5% increase for standalone.js
- Is backwards compatible: allows for either a `sheet` or `sheets` property
- `sheets` is an array (compared to an object in @davidryansoftware's fork)
- Adds a new property `sheetname` to the Sheet object; using either the `sheet` or `sheets` property, you can specific the name you want for each sheet. If a sheetname is not provided, it defaults to Excel's default naming: "Sheet1", "Sheet2", etc.
- The Relationship ID `rId` was previously hardcoded to `rId3` (not sure why it was offset by 2). After some research, I found that Excel defaults to having relationship IDs start at 1, so I reflected that change in this code

Using the new sheets property, the config would look like:
```
const config = {
    filename: 'Test',
    sheets: [
        {
            sheetname: 'Test Sheet 1',
            data: [
                [{
                    value: 'Sheet 1 Value',
                    type: 'string'
                }, {
                    value: 1234,
                    type: 'number'
                }]
            ]
        },
        {
            data: [
                [{
                    value: 'Sheet 2 Value',
                    type: 'string'
                }, {
                    value: 9876,
                    type: 'number'
                }]
            ]
        }
    ]
};
```

### CODE REVIEW

File: lib/module.js [hunk 1]
```diff
@@ -1 +1,201 @@
...tml/2006/main" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" xmlns:mv="urn:schemas-microsoft-com:mac:vml" xmlns:mx="http://schemas.microsoft.com/office/mac/excel/2008/main" xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships" xmlns:x14="http://schemas.microsoft.com/office/spreadsheetml/2009/9/main" xmlns:x14ac="http://schemas.microsoft.com/office/spreadsheetml/2009/9/ac" xmlns:xm="http://schemas.microsoft.com/office/excel/2006/main"><sheetData>{placeholder}</sheetData></worksheet>';const generateXMLWorksheet=e=>{const o=generatorRows(e);return templateSheet.replace("{placeholder}",o)};var zipcelx=e=>{if(!validator(e))throw new Error("Validation failed.");const o=new JSZip,t=o.folder("xl");t.file("workbook.xml",workbookXML),t.file("_rels/workbook.xml.rels",workbookXMLRels),o.file("_rels/.rels",rels),o.file("[Content_Types].xml",contentTypes);const s=generateXMLWorksheet(e.sheet.data);return t.file("worksheets/sheet1.xml",s),o.generateAsync({type:"blob",mimeType:"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"}).then(o=>{FileSaver.saveAs(o,`${e.filename}.xlsx`)})};export default zipcelx;export{generateXMLWorksheet};
```
Comment (elliotstoner) [inline lib/module.js:1]: I couldn't figure out why module.js was kept in its pretty form - I tried running rollup separately and it just wouldn't flatten it.

File: src/validator.js [hunk 2]
```diff
@@ -20,15 +23,52 @@ export default (config) => {
     return false;
   }
 
-  if (!Array.isArray(config.sheet.data)) {
-    console.error(INVALID_TYPE_SHEET);
+  if (config.sheet && config.sheets) {
```
Comment (elliotstoner) [inline src/validator.js:26]: The compiled file size can drop down to an 18% increase (7.1KB) if we abandon backwards-compatibility; a large chunk of the size increase is due to double validations for two different (but similar) properties... The actual code for making it backwards compatible is a single line https://github.com/egeriis/zipcelx/pull/58/files#diff-ac2318f37a52dfd510e54c3c55ec4120R20.

File: src/validator.js [hunk 3]
```diff
@@ -20,15 +23,52 @@ export default (config) => {
...
+    }
+
+    if (!childValidator(config.sheet.data)) {
+      console.error(INVALID_TYPE_SHEET_DATA);
+      return false;
+    }
+  }
+
+  if (config.sheets) {
+    const sheets = Object.keys(config.sheets);
```
Comment (elliotstoner) [inline src/validator.js]: I just realized I forgot to remove this old line of code - will update shortly

**Comment by ghost** [comment]
@elliotstoner Nice Work. I actually agree that sheets should be an array, The code I was working with at the time I created the fork was expecting an object.

File: __test__/multiSheetConfig.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/**
+  Basic working multi-sheet config
+*/
+
+export default {
+  filename: 'report',
+  sheets: [
+    {
+      sheetname: 'testSheet',
```
Comment (elliotstoner) [inline __test__/multiSheetConfig.js]: I originally kept `sheetname` all lower-case to follow `filename` being all lowercase (I imagine it as two words, thus I would have expected `fileName`) - I realize now that "filename" as one word is a valid word (against my personal preference), and thus `filename` is still following camel-case rules. I should probably update `sheetname` to be `sheetName`?

**Comment by elliotstoner** [comment]
Just thought of this - if we like this approach, do we want to have the goal of ultimately getting rid of `sheet` in favor of `sheets` in a future major release? If so, do we want to introduce a deprecation warning for `sheet` now? I would also update the docs to reflect any decisions.

File: __test__/multiSheetConfig.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/**
+  Basic working multi-sheet config
+*/
+
+export default {
+  filename: 'report',
+  sheets: [
+    {
+      sheetName: 'testSheet',
```
Comment (egeriis) [inline __test__/multiSheetConfig.js:9]: From your PR desc, it looks like you don't require `sheetName`. If that's correct, mind adding a test case for that?

Comment (elliotstoner) [reply __test__/multiSheetConfig.js:9]: That's correct! I went ahead and added a second "sheet" that doesn't have a sheetName and squashed the commits into one

**Comment by elliotstoner** [comment]
Bump @egeriis. It's been almost 9 months - do we want to move forward with this or are we abandoning it? If the latter then I'm going to branch this off onto my own package

**Comment by elliotstoner** [comment]
This repo seems unmaintained. I'm going to close this pull request in favor of migrating to [xljsx-lite](https://github.com/elliotstoner/xljsx-lite)! For anyone looking for this functionality, feel free to check out my repo with these improvements, with more to come! There's two versions: v1.0.2 is the closest to this current repo, while v2.0.0 has some additional improvements (automatic typing, even further reduced bundle size). I look forward to feedback about how we can improve this together!

**Pull request closed without merging**